### PR TITLE
PR #1108 with issue #1122

### LINKED ISSUE #1122: Remove PairingRDD
After #1108 merges, `PairingRDD` will be unused and should be removed.

### PR DESCRIPTION
**Pull request by fnothaft**
This is @akmorrow13's PR #1108 with code tacked on to resolve #1122. @akmorrow13, I figured it might just be easier to do it this way. I rebased this on both ToT and your latest coverage branch, so it should be your latest and greatest coverage code.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala [hunk 1]
```diff
@@ -150,9 +150,25 @@ object ReferenceRegion {
     }
   }
 
+  /**
+   * Extracts ReferenceRegion from Feature
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala]: doc lines should be full sentences

File: adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala [hunk 2]
```diff
@@ -150,9 +150,25 @@ object ReferenceRegion {
...
+   *
+   * @param feature Feature to extract ReferenceRegion from
+   * @return Extracted ReferenceRegion
+   */
   def apply(feature: Feature): ReferenceRegion = {
     new ReferenceRegion(feature.getContigName, feature.getStart, feature.getEnd)
   }
+
+  /**
+   * Extracts ReferenceRegion from Coverage
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala]: same here

**Comment by heuermh** [comment]
LGTM, other than two small doc nits.

I'm happy with how the feature <--> converage conversions worked out.

**Comment by fnothaft** [comment]
Pulled in @akmorrow13 's latest changes, and rebased on master.

**Comment by fnothaft** [comment]
@heuermh this resolves your comments.

**Comment by heuermh** [comment]
Thank you, @fnothaft @akmorrow13!

**Pull request merged**